pkg/support/dump.go

```diff
@@ -221,7 +223,7 @@ func directoryCopy(fs afero.Fs, targetDir string, srcDir string, prefixFilter st
 // writeFile writes the given data to the specified filePath. Param "resource" is used to identify
 // the type of the given data in the error message.
 func writeFile(fs afero.Fs, filePath string, resource string, data []byte) error {
-	err := afero.WriteFile(fs, filePath, data, 0644)
+	err := writeFileFunc(fs, filePath, data, 0644)
```

Comment:
The reason why it calls `afero.WriteFile(fs, filePath, data, 0644)` instead of `os.WriteFile` is to abstract the filesystem interface so it can be easily mocked with a fake filesystem, with which the content of written file can also be checked.
Please search `afero.NewMemMapFs()` for example.

Reply:
Thanks for the info. Updated